2018/f8801_2019.py

```diff
@@ -0,0 +1,90 @@
+from form import Form, FilingStatus
+
+class F8801_2019(Form):
+    """Form 8801, Credit for Prior Year Minimum Tax
+
+    This form is for 2019, but since it depends on values
+    from 2018, it is generated along with 2018's
+    forms. Line 21 from this form must be manually entered into
+    2019's input dictionary.
+    """
+    def __init__(f, inputs, f1040, f6251, f8801, sched_d):
+        super(F8801_2019, f).__init__(inputs)
+        f['1'] = f6251.rowsum(['1', '2e'])
+        f['2'] = f6251.rowsum(['2a', '2b', '2c', '2d', '2g', '2h'])
+        # TODO: Line 3, minimum tax credit net operating loss deduction
+        f['4'] = max(0, f['1'] + f['2'] + f['3'])
+        assert(inputs['status'] != FilingStatus.SEPARATE or f['4'] <= 718800)
+        f.comment['15'] = 'Net minimum tax on exclusion items'
+        if f['4'] == 0:
+            f['15'] = 0
+        else:
+            f['5'] = f6251.EXEMPTIONS[inputs['status']]
+            f['6'] = f6251.EXEMPT_LIMITS[inputs['status']]
+            f['7'] = max(0, f['4'] - f['6'])
+            f['8'] = f['7'] * 0.25
+            f['9'] = max(0, f['5'] - f['8'])
+            f['10'] = max(0, f['4'] - f['9'])
+            if f['10'] == 0:
+                f['15'] = 0
+            else:
+                # TODO: form 2555
+                if (f1040['13'] and not sched_d.mustFile()) or f1040['9b'] or \
+                        (sched_d['15'] > 0 and sched_d['16'] > 0):
+                    f['27'] = f.get('10')
+                    # TODO: Schedule D tax worksheet
+                    assert(not sched_d['18'] and not sched_d['19'])
+                    cg_worksheet = f1040.div_cap_gain_tax_worksheet(inputs,
+                                                                    sched_d)
+                    f['28'] = cg_worksheet['6']
+                    f['30'] = f.get('28')
+                    f['31'] = min(f['27'], f['30'])
+                    f['32'] = f['27'] - f['31']
+                    f['33'] = f6251.amt(inputs['status'], f['32'])
+                    f['34'] = f1040.BRACKET_LIMITS[inputs['status']][1]
+                    f['35'] = cg_worksheet['7']
+                    f['36'] = max(0, f['34'] - f['35'])
+                    f['37'] = min(f['27'], f['28'])
+                    f['38'] = min(f['36'], f['37'])
+                    f['39'] = f['37'] - f['38']
+                    f['40'] = f1040.BRACKET_LIMITS[inputs['status']][-1]
```

Comment:
BRACKET_LIMITS should be replaced with CAPGAIN20_LIMITS on this line.

Reply:
Hmm,
1) I'm not seeing exactly how this works. These refer to status on 1040-NR, right? If you understand this better than me, it'd be great if you could confirm what it should be.
2) I copied this from 2017/f8801_2018.py. That one uses BRACKET_LIMITS. Does that need to be fixed too?

Reply:
inputs['status'] is just your filing status.

In 2017, these thresholds were pegged to the regular tax brackets, but in 2018 with the tax law changes, the capital gain thresholds are now called out separately. You can see the same difference if you look at 2017/f6251.py versus 2018/f6251.py.

Anyways, f['34'] should be f1040.CAPGAIN15_LIMITS[inputs['status']] and f['40'] should be f1040.CAPGAIN20_LIMITS[inputs['status']]

Reply:
Thanks, done.